v03_pipeline/lib/reference_datasets/reference_dataset.py

```diff
@@ -0,0 +1,52 @@
+from enum import Enum
+
+from v03_pipeline.lib.model import AccessControl, DatasetType, Env, ReferenceGenome
+from v03_pipeline.lib.reference_data.config import CONFIG
+
+
+class ReferenceDataset(str, Enum):
+    cadd = 'cadd'
+    hgmd = 'hgmd'
+
+    @property
+    def _config(self) -> dict:
+        return CONFIG[self.value]
+
+    @classmethod
+    def for_reference_genome_dataset_type(
+        cls,
+        reference_genome: ReferenceGenome,
+        dataset_type: DatasetType,
+    ) -> list['ReferenceDataset']:
+        reference_datasets = {
+            (ReferenceGenome.GRCh37, DatasetType.SNV_INDEL): [
+                ReferenceDataset.cadd,
+                ReferenceDataset.hgmd,
+            ],
+            (ReferenceGenome.GRCh38, DatasetType.SNV_INDEL): [
+                ReferenceDataset.cadd,
+                ReferenceDataset.hgmd,
+            ],
+        }.get((reference_genome, dataset_type), [])
```

Comment:
I wonder if it would be clearer to add a `DATASET_TYPES`  to the config and then just iterate through here to get them. That way we do not need to explicitly configure them on/ off in a separate place from where they are configured
```
CONFIG = {
     cadd: {
            ReferenceGenome.GRCh37: {
                VERSION: '0.0',
                DATASET_TYPES: DatasetType.SNV_INDEL,
            },
}
...
def for_reference_genome_dataset_type(cls, reference_genome: ReferenceGenome, dataset_type: DatasetType) -> list['ReferenceDataset']:
    reference_datasets = [
        dataset for dataset, config in CONFIG.items()
        if dataset_type in config.get(reference_genome, {}).get(DATASET_TYPES, [])
    ] 
    ...
```

Reply:
I implemented it this way! I like that it keeps the config for a reference dataset in one place.